pypeit/core/telluric.py

```diff
@@ -0,0 +1,1259 @@
+    # Interpolate PCA components onto wave_grid
+    pca_interp = scipy.interpolate.interp1d(wave_pca_c*(1.0 + redshift),pca_comp_c, bounds_error=False, fill_value=0.0, axis=1)
+    pca_comp_new = pca_interp(wave_grid)
+    # Generate a mixture model for the coefficients prior, what should ngauss be?
+    prior = mixture.GaussianMixture(n_components = npca-1).fit(coeffs_c[:, 1:npca])
+    # Construct the PCA dict
+    pca_dict = {'npca': npca, 'components': pca_comp_new, 'prior': prior, 'coeffs': coeffs_c,
+                'z_fid': redshift, 'dloglam': dloglam}
+    return pca_dict
+
+def pca_eval(theta,pca_dict):
+    # theta_pca[0] is redshift
+    # theta_pca[1] is norm
+    # theta_pca[2:npca+1] are the PCA dimensionality
+
+    C = pca_dict['components']
+    z_fid = pca_dict['z_fid']
+    dloglam = pca_dict['dloglam']
+    npca = pca_dict['npca']  # Size of the PCA currently being used, original PCA in the dict could be larger
+    z_qso = theta[0]
+    norm = theta[1]
+    A = theta[2:]
+    dshift = int(np.round(np.log10((1.0 + z_qso)/(1.0 + z_fid))/dloglam))
+    C_now = np.roll(C[:npca,:], dshift, axis=1)
+    return norm*np.exp(np.dot(np.append(1.0,A),C_now))
+
+def pca_lnprior(theta,pca_dict):
+    gaussian_mixture_model = pca_dict['prior']
+    A = theta[2:]
+    return gaussian_mixture_model.score_samples(A.reshape(1,-1))
+
+
+def read_telluric_grid(filename, wave_min=None, wave_max=None, pad = 0):
+
+    hdul = fits.open(filename)
+    wave_grid_full = 10.0*hdul[1].data
+    model_grid_full = hdul[0].data
+    nspec_full = wave_grid_full.size
+
+    if wave_min is not None:
+        ind_lower = np.argmin(np.abs(wave_grid_full - wave_min)) - pad
+    else:
+        ind_lower = 0
+    if wave_max is not None:
+        ind_upper = np.argmin(np.abs(wave_grid_full - wave_max)) + pad
+    else:
+        ind_upper=nspec_full
+    wave_grid = wave_grid_full[ind_lower:ind_upper]
+    model_grid = model_grid_full[:,:,:,:, ind_lower:ind_upper]
+
+    pg = hdul[0].header['PRES0']+hdul[0].header['DPRES']*np.arange(0,hdul[0].header['NPRES'])
+    tg = hdul[0].header['TEMP0']+hdul[0].header['DTEMP']*np.arange(0,hdul[0].header['NTEMP'])
+    hg = hdul[0].header['HUM0']+hdul[0].header['DHUM']*np.arange(0,hdul[0].header['NHUM'])
+    if hdul[0].header['NAM'] > 1:
+        ag = hdul[0].header['AM0']+hdul[0].header['DAM']*np.arange(0,hdul[0].header['NAM'])
+    else:
+        ag = hdul[0].header['AM0']+1*np.arange(0,1)
+
+    dwave, dloglam, resln_guess, pix_per_sigma = wvutils.get_sampling(wave_grid)
+    tell_pad_pix = int(np.ceil(10.0 * pix_per_sigma))
+
+    tell_dict = dict(wave_grid=wave_grid, dloglam=dloglam,
+                     resln_guess=resln_guess, pix_per_sigma=pix_per_sigma, tell_pad_pix=tell_pad_pix,
+                     pressure_grid=pg, temp_grid=tg, h2o_grid=hg, airmass_grid=ag, tell_grid=model_grid)
+    return tell_dict
+
+
+def interp_telluric_grid(theta,tell_dict):
+
```

Comment:
doc string.  I'll stop repeating it.  Obviously
this is pervasive.  Maybe we should remove this
module altogether.